plugin/src/main/java/org/gradle/testretry/internal/visitors/SpockParameterClassVisitor.java

```diff
@@ -80,6 +95,25 @@ private static String normalizeMethodName(String methodPattern) {
         return methodPattern.replaceAll(SPOCK_PARAM_PATTERN, PARAM_PLACEHOLDER);
     }
 
+    @Override
+    public void visit(int version, int access, String name, String signature, String superName, String[] interfaces) {
+        super.visit(version, access, name, signature, superName, interfaces);
+        if (superName != null && !superName.equals("java/lang/Object")) {
+            final FileCollection collection = spec.getTestClassesDirs().filter(d -> Files.exists(parentClass(d, superName)));
```

Comment:
can we even make better use of the gradle JvmTestExecutionSpec (eg `JvmTestExecutionSpec.getPreviousFailedTestClasses()`). The `org.gradle.api.internal.tasks.testing.junit.JUnitDetector` knows already about the superClasses